scalding-core/src/main/scala/com/twitter/scalding/TypedDelimited.scala

```diff
@@ -29,19 +32,26 @@ trait TypedSeperatedFile extends Serializable {
   def skipHeader: Boolean = false
   def writeHeader: Boolean = false
 
-  def apply[T: Manifest: TupleConverter: TupleSetter](path: String): FixedPathTypedDelimited[T] =
+  def apply[T: TypeDescriptor](path: String): TypedTextDelimited[T] =
     apply(Seq(path))
 
-  def apply[T: Manifest: TupleConverter: TupleSetter](paths: Seq[String]): FixedPathTypedDelimited[T] = {
-    val f = Dsl.intFields(0 until implicitly[TupleConverter[T]].arity)
-    apply(paths, f)
+  def apply[T: TypeDescriptor](paths: Seq[String]): TypedTextDelimited[T] = {
+    new FixedTypedText(source.TypedSep(separator), paths: _*)
   }
 
-  def apply[T: Manifest: TupleConverter: TupleSetter](path: String, f: Fields): FixedPathTypedDelimited[T] =
+  def apply[T: TypeDescriptor](path: String, f: Fields): TypedTextDelimited[T] =
     apply(Seq(path), f)
 
-  def apply[T: Manifest: TupleConverter: TupleSetter](paths: Seq[String], f: Fields): FixedPathTypedDelimited[T] =
-    new FixedPathTypedDelimited[T](paths, f, skipHeader, writeHeader, separator)
+  def apply[T: TypeDescriptor](paths: Seq[String], f: Fields): TypedTextDelimited[T] = {
+    val td = implicitly[TypeDescriptor[T]]
+    require(f.size == td.fields.size)
+    val newTd = new TypeDescriptor[T] {
+      def setter = td.setter
+      def converter = td.converter
+      def fields = f
```

Comment:
see: https://github.com/Cascading/cascading/blob/3.0/cascading-core/src/main/java/cascading/tuple/Fields.java

I think we want:

```scala
val newFields = new Fields(f.get, td.fields.getTypes)
```
we don't want to lose the types, just keep the names.